File: Types of Search/Binary search.py
def binary_search(start,end,a,k):
    for i in a:
        mid=(start+end)//2
        if a[mid]==k:
            print(mid)
            break
        elif a[mid]>k:
            return binary_search(start,mid,a,k)
            break
        else:
            return binary_search(mid+1,end,a,k)
n=int(input('enter a number of element:'))
a=[]
for i in range(n):
    a.append(int(input('enter a items:')))
for i in range(0,n):
    for j in range(0,n-i-1):
        if a[j]>a[j+1]:
            a[j],a[j+1]=a[j+1],a[j]
print('the sorted list is :',a)

k=int(input('enter a search item:'))

print(binary_search(0,n-1,a,k))

File: Types of Search/binary search.c
#include<stdio.h>
int main()
{
    int n,i,a[100],e,found=0,first,last,mid,min,j,t,k;
    printf("\ncreate the array");
    printf("\nenter the limit:");
    scanf("%d",&n);
    for(i=0;i<n;i+=1)
    {
        printf("\nenter the value of a[%d]:",i);
        scanf("%d",&a[i]);
    }
    printf("\n ARRAY\n");
    
    for(i=0;i<n;i+=1)
    {
        printf("%d\t",a[i]);
    }
    //selection sort 
    for(i=0;i<n;i+=1)
    {
        min=i;
        for(j=i+1;j<n;j+=1)
        {
            if(a[min]>a[j])
            {
                min=j;
            }
            if(min!=i)
            {
                t=a[min];
                a[min]=a[i];
                a[i]=t;
            }
        }
    }  

    
    printf("\n sorted ARRAY\n");
    for(k=0;k<n;k+=1)
    {
        printf("%d\t",a[k]);
    }
    printf("\nenter the search element:");
    scanf("%d",&e);
    first=0;
    last=n-1;
    mid=(first+last)/2;
    while(first<=last)
    {
        if(e==a[mid])
        {
            printf("element found at index %d",mid);
            found+=1;
            break;
        }
        else if(e<a[mid])
        {
            last=mid-1;
        }
        else
        {
            first=mid+1;
        }
        mid=(first+last)/2;

    }
    if(found==0)
    {
    	printf("\nnot found");
	}
	
}

